### PR TITLE
fix doc of ensure-rep-shown and add check for it

### PR DESCRIPTION
**Pull request by Syntacticlosure**

### CODE REVIEW

File: drracket/drracket/private/unit.rkt [hunk 1]
```diff
@@ -2355,6 +2355,9 @@
           (toggle-show/hide-definitions)
           (update-shown)))
       (define/public (ensure-rep-shown rep)
+        (unless (is-a? rep drracket:rep:text<%>)
+           (error 'ensure-rep-shown
```
Comment (sorawee) [inline drracket/drracket/private/unit.rkt:2359]: The documentation of `error` states:

> When possible, use functions such as `raise-argument-error`, instead, which construct messages that follow Racket’s error message conventions.

So let's use `raise-argument-error` instead.

Comment (sorawee) [reply drracket/drracket/private/unit.rkt:2359]: Oh, and thanks to @rfindler for catching this!

**Comment by rfindler** [comment]
Something like this, maybe?

```
(raise-argument-error
  'ensure-rep-shown
  (format "~s" '(is-a?/c drracket:rep:text<%>))
  rep)
```

**Comment by rfindler** [comment]
And also squash the commits. I'm happy to do this if you want, @Syntacticlosure ?

**Comment by spdegabrielle** [comment]
I think this is just a button press now? https://github.blog/2016-04-01-squash-your-commits/

**Comment by samth** [comment]
It is, but it would cause the commit to be authored by @rfindler.

**Comment by sorawee** [comment]
Coauthored I think, just like https://github.com/racket/racket/commit/ee73a550aabd1f6ed72d90f00dddb6915479fc52.

**Comment by rfindler** [comment]
I don't see a commit that fixes the error call. Am I just looking in the wrong place?

I am happy to not have my name on this commit (or to have it; either is fine).

**Comment by Syntacticlosure** [comment]
@rfindler 
It has been a long time since i sent this pull request, so i almost forget it.
It's OK to squash my commits even that it will not be authored by me.

**Comment by rfindler** [comment]
Thank you! And sorry for taking so long to get around to this.

**Pull request closed without merging**